python/sparkdl/transformers/tf_tensor.py

```diff
@@ -62,91 +65,84 @@ def setParams(self, tfInputGraph=None, inputMapping=None, outputMapping=None, tf
-                if orig_tensor.dtype == tf.float64:
-                    output_tensor = tf.identity(spark_placeholder)
-                # Otherwise, cast the Spark input to the datatype of the original tensor
-                else:
-                    output_tensor = tf.cast(spark_placeholder, dtype=orig_tensor.dtype)
-                input_names.append(spark_placeholder.name)
-                output_names.append(output_tensor.name)
-
-        # Concatenate our injected-op subgraph with the original graph and return the result.
-        injected_op_subgraph_fn = GraphFunction(graph_def=injected_op_subgraph.as_graph_def(),
-                                             input_names=input_names, output_names=output_names)
-        return GraphFunction.fromList([("sparkdl_ops", injected_op_subgraph_fn),
-                                      ("user_graph", user_graph_fn)])
+            with tf.name_scope(self.SPARKDL_OP_SCOPE):
+                for _, orig_tensor_name in self.getInputMapping():
+                    orig_tensor = tfx.get_tensor(orig_tensor_name, user_graph)
+                    # Create placeholder with same shape as original input tensor, but that accepts
+                    # float64 input from Spark.
+                    spark_placeholder = tf.placeholder(tf.float64, shape=orig_tensor.shape,
+                                                       name=tfx.op_name(orig_tensor_name))
+                    # If the original tensor was of type float64, just pass through the Spark input
+                    if orig_tensor.dtype == tf.float64:
+                        input_map[orig_tensor_name] = spark_placeholder
+                    # Otherwise, cast the Spark input to the datatype of the original tensor
+                    else:
+                        input_map[orig_tensor_name] = tf.cast(spark_placeholder,
+                                                              dtype=orig_tensor.dtype)
+            tf.import_graph_def(graph_def=user_graph_def, input_map=input_map, name="")
+        return injected_op_subgraph.as_graph_def(add_shapes=True)
 
     def _optimize_for_inference(self):
         """ Optimize the graph for inference """
         gin = self.getTFInputGraph()
-        input_mapping = self.getInputMapping()
-        output_mapping = self.getOutputMapping()
-        input_node_names = [tfx.op_name(tnsr_name) for _, tnsr_name in input_mapping]
-        output_node_names = [tfx.op_name(tnsr_name) for tnsr_name, _ in output_mapping]
-
-        # Build GraphFunction wrapping the model graph
-        graph_fn = GraphFunction(graph_def=gin.graph_def, input_names=input_node_names,
-                                 output_names=output_node_names)
         # Inject cast ops to convert float64 input fed from Spark into the datatypes of the
         # Graph's input nodes.
-        graph_fn_with_casts = self._addCastOps(graph_fn)
+        graphdef_with_casts = self._addCastOps(gin.graph_def)
```

Comment:
nit: just oneline it since `gin` is not used otherwise:
  graphdef_with_casts = self._addCastOps(self.getTFInputGraph().graph_def)